includes/admin/debug-tools/class-wcs-notifications-debug-tool-processor.php

```diff
@@ -0,0 +1,327 @@
+
+		$subscriptions_notifications = WC_Subscriptions_Core_Plugin::instance()->notifications_scheduler;
+
+		foreach ( $batch as $subscription_id ) {
+			$subscription = wcs_get_subscription( $subscription_id );
+
+			if ( WC_Subscriptions_Email_Notifications::notifications_globally_enabled() ) {
+				$subscriptions_notifications->update_status( $subscription, $subscription->get_status(), null );
+			} else {
+				$subscriptions_notifications->unschedule_all_notifications( $subscription );
+			}
+
+			// Update the subscription's update time to mark it as updated.
+			$subscription->set_date_modified( time() );
+			$subscription->save();
+		}
+
+		// Update tool state.
+		$state                = $this->get_tool_state();
+		$state['last_offset'] = isset( $state['last_offset'] ) ? absint( $state['last_offset'] ) + count( $batch ) : count( $batch );
+		$this->update_tool_state( $state );
+	}
+
+	/**
+	 * Default (preferred) batch size to pass to 'get_next_batch_to_process'.
+	 * The controller will pass this size unless it's externally configured
+	 * to use a different size.
+	 *
+	 * @return int Default batch size.
+	 */
+	public function get_default_batch_size(): int {
+		return 20;
+	}
+
+	/**
+	 * Start the background process for batch processing subscription notifications updates.
+	 *
+	 * @return string Informative string to show after the tool is triggered in UI.
+	 */
+	public function enqueue(): string {
+		$batch_processor = WCS_Batch_Processing_Controller::instance();
+		if ( $batch_processor->is_enqueued( self::class ) ) {
+			return __( 'Background process for updating subscription notifications already started, nothing done.', 'woocommerce-subscriptions' );
+		}
+
+		$batch_processor->enqueue_processor( self::class );
+
+		return __( 'Background process for updating subscription notifications started', 'woocommerce-subscriptions' );
+	}
+
+	/**
+	 * Stop the background process for batch processing subscription notifications updates.
+	 *
+	 * @return string Informative string to show after the tool is triggered in UI.
+	 */
+	public function dequeue(): string {
+		$batch_processor = WCS_Batch_Processing_Controller::instance();
+		if ( ! $batch_processor->is_enqueued( self::class ) ) {
+			return __( 'Background process for updating subscription notifications not started, nothing done.', 'woocommerce-subscriptions' );
+		}
+
+		$batch_processor->remove_processor( self::class );
+		return __( 'Background process for updating subscription notifications stopped', 'woocommerce-subscriptions' );
+	}
+
+	/**
+	 * Add the tool to start or stop the background process that manages notification batch processing.
+	 *
+	 * @param array $tools Old tools array.
+	 * @return array Updated tools array.
+	 */
+	public function handle_woocommerce_debug_tools( array $tools ): array {
+
+		if ( ! WC_Subscriptions_Email_Notifications::notifications_globally_enabled() ) {
+			$tools['start_add_subscription_notifications'] = array(
```

Comment:
❓ Just so I understand, this specific tool is added as a "stub" of sorts to the **WooCommerce → Status → Tools** even though the feature is disabled just for reference? It will always be disabled when the feature isn't enabled.

Reply:
Yes, that's the idea. Instead of completely hiding this, we keep the UI consistent and use friendly messages. 

Initially, before the mass unscheduling, this was (kinda) necessary in case the progressing unscheduling failed. So, essentially, the same handling was maintained after the improvement. 

Perhaps the copy could use some improvement? I added some info in [this PR](https://github.com/Automattic/woocommerce-subscriptions-core/pull/704) if you'd like to take a look. 

![Screenshot 2024-10-21 at 5 09 21 PM](https://github.com/user-attachments/assets/d9965e74-8809-4cb4-a21b-e2c626fbf8dc)